index.bs

```diff
@@ -499,6 +523,64 @@ Modify the <a spec=HTML>downloads a hyperlink</a> algorithm to ensure that downl
   1. If [=this=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then return.
 </div>
 
+<h3 id="permissions">Waiting before permissions checks</h3>
+
+Many specifications need to be patched so that, if a given algorithm invoked in a [=prerendering browsing context=], most of its work is deferred until the browsing context is [=prerendering browsing context/activated=]. This is tricky to do uniformly, as many of these specifications do not have great hygeine around <a href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-for-spec-authors">using the event loop</a>. Nevertheless, the following sections give our best attempt.
+
+<h4 id="patch-geolocation">Geolocation API</h4>
+
+<div algorithm="Geolocation getCurrentPosition patch">
+  Modify the {{Geolocation/getCurrentPosition()}} method steps by prepending the following step:
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then append the following steps to [=this=]'s [=platform object/post-prerendering activation steps list=] and return.
+</div>
+
+<div algorithm="Geolocation watchPosition patch">
+  Modify the {{Geolocation/watchPosition()}} method steps by prepending the following step:
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then:
+
+    1. Let |watchProcessId| be a new unique <a spec=GEOLOCATION>watch process</a> ID, and note it as a <dfn>post-prerendering activation geolocation watch process ID</dfn>.
+
+    1. Append the following steps to [=this=]'s [=platform object/post-prerendering activation steps list=], with the modification that the <a spec=GEOLOCATION>watch process</a> generated must use |watchProcessId| as its ID, and return |watchProcessId|.
```

Comment:
This is slightly confusing in the subtlety of the change it applies to another step. Maybe that's fine for now because we want to keep the delta small, but I assume if this were to be merged we would refactor this?

Reply:
If we had infinite time, it might even be worth having the upstream thing have an explicit map of watch IDs to some structure that includes an abort signal (even if it's not web-exposed). But that's probably too much of a detour to ask for right now. :)

Reply:
Yeah, the geolocation spec is so old-school, I'm not sure that even if we did incorporate it, I'd refactor as you suggest. If you look at how watch processes are currently associated and IDs are generated, it's so hand-wavy, that making this one thing precise seems like it might be more confusing than helpful.

I am tempted to refactor the geolocation spec entirely, but I feel like I'd quickly run into interop issues and then get stuck waiting for implementers to come agree on something.